src/factory.ts

```diff
@@ -28,26 +28,98 @@ import {DEFAULT_LABELS} from './constants';
 // eslint-disable-next-line @typescript-eslint/no-var-requires
 const parseGithubRepoUrl = require('parse-github-repo-url');
 
+// types defining methods available to call on instances
+export type ReleasePRMethod = 'run' | 'latestTag';
+export type GitHubReleaseMethod = 'run';
+export type Method = ReleasePRMethod | GitHubReleaseMethod;
+
+// types defining cli commands and their options
+export type ReleasePRCommands = 'release-pr' | 'latest-tag';
+export type GitHubReleaseCommands = 'github-release';
+type Command = ReleasePRCommands | GitHubReleaseCommands;
+type IsReleasePRCmd = {
+  command: Command;
+  options: ReleasePRFactoryOptions;
+};
+type IsGitHubReleaseCmd = {
+  command: Command;
+  options: GitHubReleaseFactoryOptions;
+};
+
+// shorthand aliases for instance/method call return types
+export type ReleasePRRunResult = Promise<number | GitHubTag | undefined>;
+export type GitHubReleaseRunResult = Promise<GitHubReleaseResponse | undefined>;
+export type RunResult = ReleasePRRunResult | GitHubReleaseRunResult;
+
+function isGitHubReleaseCmd(
+  cmdOpts: IsReleasePRCmd | IsGitHubReleaseCmd
+): cmdOpts is IsGitHubReleaseCmd {
+  const {command, options} = cmdOpts;
+  return command === 'github-release' && typeof options === 'object';
+}
+
+function isReleasePRCmd(
+  cmdOpts: IsReleasePRCmd | IsGitHubReleaseCmd
+): cmdOpts is IsReleasePRCmd {
+  const {command, options} = cmdOpts;
+  return (
+    (command === 'release-pr' || command === 'latest-tag') &&
+    typeof options === 'object'
+  );
+}
+
 function runCommand(
-  command: string,
+  command: ReleasePRCommands,
+  options: ReleasePRFactoryOptions
+): ReleasePRRunResult;
+function runCommand(
+  command: GitHubReleaseCommands,
+  options: GitHubReleaseFactoryOptions
+): GitHubReleaseRunResult;
+function runCommand(
+  command: Command,
   options: GitHubReleaseFactoryOptions | ReleasePRFactoryOptions
-): Promise<number | undefined | ReleaseResponse> {
-  if (isGitHubRelease(command, options)) {
-    return factory.run(githubRelease(options));
+): RunResult {
+  let result: RunResult;
+  const cmdOpts = {command, options};
+  if (isReleasePRCmd(cmdOpts)) {
+    let method: ReleasePRMethod = 'run';
+    if (command === 'latest-tag') {
+      method = 'latestTag';
+    }
+    result = factory.call(releasePR(cmdOpts.options), method);
+  } else if (isGitHubReleaseCmd({command, options})) {
+    result = factory.call(githubRelease(cmdOpts.options), 'run');
```

Comment:
I augmented the `is{GitHubRelease,ReleasePR}Cmd` functions' type predicates specifically to be able to narrow the typing such that `cmdOpts.options` is the correct type going into `releasePR()` and `githubRelease()`